corgidrp/l2a_to_l2b.py

```diff
@@ -171,13 +171,33 @@ def correct_bad_pixels(input_dataset):
         not want to correct everything in the DQ extension
         - Different bad pixels in the DQ may be corrected differently.
 
-
     Args:
         input_dataset (corgidrp.data.Dataset): a dataset of Images (L2a-level)
+        bp_mask (corgidrp.data.BadPixelMap): Bad-pixel mask flagging all bad pixels
+            in that frame. Must be 0 (good) or 1 (bad) at every pixel.
 
     Returns:
-        corgidrp.data.Dataset: a version of the input dataset with bad pixels corrected
-    """
-
-    return None
-
+        corgidrp.data.Dataset: a version of the input dataset with bad detector
+        pixels and cosmic rays replaced by 0s
+ 
+    """
+
+    data = input_dataset.copy()
+    data_cube = data.all_data
+    dq_cube = data.all_dq
+
+    for i in range(data_cube.shape[0]):
+        # load CR mask
+        cr_mask = dq_cube[i]
+        # combine CR and BP masks
+        bp_cr_mask = (cr_mask+bp_mask[0].data).astype(int)
+        # mask affected pixels with NaN
+        bp = np.where(bp_cr_mask != 0)
+        data_cube[i, bp[0], bp[1]] = np.nan
+        dq_cube[i] = bp_cr_mask
+
```

Comment:
Right now in the implementation doc, we're using the 1st bit to signify that a pixel has been replaced. Although in this case, since it's a nan, it'll be obvious I think we still probably want to update that bit in the DQ too.

Reply:
Not sure I follow. The image data pixels that are replaced by NaN are those that have a non-zero bit in their DQ. That is, they already have it. The updated DQ value for each pixel on the image data happens a step earlier, not in these lines.

Reply:
The DQ has different bits that indicate different things. One of the bits (bit 1) indicates that a pixel's value has been replaced. Since we're replacing it here, then I think we should set that bit.

Reply:
@maxwellmb: Is it not set already? 

`# combine DQ and BP masks`
  `bp_dq_mask_bit = np.bitwise_or(dq_mask_bit, bp_mask_bit)`
        `# Transform back to integer DQ values`
        `bp_dq_mask = np.packbits(bp_dq_mask_bit).reshape(dq_mask.shape[0], dq_mask.shape[1])`
       ` # mask affected pixels with NaN`
        `bp = np.where(bp_dq_mask != 0)`

I do not see why DQ values need to be set after the last line if they are already set to a non-good pixel value. That's what is used to replace a pixel on the image data by NaN. What would be the new setting for the DQ pixel? It should be the same non-zero value. DQ values are not bitarrays. They are integers, and you asked me to combine them as bitarrays for operations, only. I'm also confused as to what bit 1 means (first?, or second if we count first index as 0). Can you add an example of what you would do because I'm not getting it? Thanks a lot!